Solutions/GoogleCloudPlatformIAM/Analytic Rules/GCPIAMHighPrivilegedRoleAdded.yaml

```diff
@@ -0,0 +1,33 @@
+id: 86112c4b-2535-4178-aa0e-ed9e32e3f054
+name: GCP IAM - High privileged role added to service account
+description: |
+  'Detects when high privileged role was added to service account.'
+severity: High
+requiredDataConnectors:
+  - connectorId: GCPIAMDataConnector
+    dataTypes:
+      - GCP_IAM
+queryFrequency: 15m
+queryPeriod: 15m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1078
+query: |
+  let privileged_roles = dynamic(['roles/iam.securityAdmin', 'roles/secretmanager.admin', 'roles/secretmanager.secretAccessor', 'roles/apigateway.admin', 'roles/logging.admin', 'roles/iam.organizationRoleAdmin', 'roles/iam.roleAdmin', 'roles/iam.serviceAccountAdmin', 'roles/iam.serviceAccountCreator', 'roles/iam.serviceAccountKeyAdmin']);
+  GCP_IAM
+  | where PayloadMethodname =~ 'SetIamPolicy'
+  | extend action = parse_json(todynamic(PayloadServicedataPolicydeltaBindingdeltas))[0]['action']
+  | where action =~ 'ADD'
+  | extend role = parse_json(todynamic(PayloadServicedataPolicydeltaBindingdeltas))[0]['role']
+  | where role in~ (privileged_roles)
+  | project-away action
+  | extend AccountCustomEntity = PayloadAuthenticationinfoPrincipalemail

```

Comment:
Please add IPCustomEntity if it is surfaced in the log. Can we also have the timestamp field mapped please (i.e. extend timestamp = TimeGenerated). Thanks